static/app/views/metrics/utils/index.tsx

```diff
@@ -1,36 +1,62 @@
-import {BooleanOperator} from 'sentry/components/searchSyntax/parser';
+import {
+  MetricSeriesFilterUpdateType,
+  type MetricsQuery,
+} from 'sentry/utils/metrics/types';
+import {MutableSearch} from 'sentry/utils/tokenizeSearch';
 
-function constructQueryString(queryObject: Record<string, string>) {
-  return Object.entries(queryObject)
-    .map(([key, value]) => `${key}:"${value}"`)
-    .join(' ');
-}
+import {addToFilter, excludeFromFilter} from '../../discover/table/cellAction';
 
+/**
+ * Updates query string with tag values from the passed groupBys.
+ */
 export function extendQueryWithGroupBys(
-  query: string = '',
-  groupBys?: (Record<string, string> | undefined)[]
+  query: string,
+  groupBys: (Record<string, string> | undefined)[]
 ) {
-  const focusedSeriesQuery = groupBys
-    ?.map(groupBy => {
-      if (!groupBy || Object.keys(groupBy).length === 0) {
-        return '';
+  const mutableSearch = new MutableSearch(query);
+
+  groupBys.forEach(groupBy => {
+    if (!groupBy) {
+      return;
+    }
+    Object.entries(groupBy).forEach(([key, value]) => {
+      if (!value) {
+        return;
       }
-      return constructQueryString(groupBy);
-    })
-    .filter(Boolean)
-    .join(` ${BooleanOperator.OR} `);
+      addToFilter(mutableSearch, key, value);
+    });
+  });
 
-  if (!focusedSeriesQuery) {
-    return query;
-  }
+  return mutableSearch.formatString();
+}
+
+/**
+ * Used when a user clicks on filter button in the series summary table. Applies
+ * tag values to the filter string of the query. Removes the tags from query groupyBy
+ */
+export function updateQueryWithSeriesFilter(
+  query: MetricsQuery,
+  groupBys: Record<string, string>,
+  updateType: MetricSeriesFilterUpdateType
+) {
+  const mutableSearch = new MutableSearch(query.query ?? '');
```

Comment:
Just as a warning, the `MutableSearch` is legacy code and it does not use the search syntax parser internally.
So, we should double check that it works with all search syntax features that we support.
Maybe add tests with more complex queries.